client/py_client/ipfs.py

```diff
@@ -29,3 +36,28 @@ def add_recursive(path_to_files):
     @staticmethod
     def add_recursive_multiple(paths):
         return [Ipfs.add_recursive(f) for f in paths]
+
+    @staticmethod
+    def add_recursive_remote(path_to_files, IPFS_API_KEY):
+        if IPFS_API_KEY != '':
+            ret = subprocess.run(
+                ["curl", "-X", "POST", "-F", f"file=@{path_to_files}", IPFS_API_KEY, "https://ipfs.infura.io:5001/api/v0/add"], stdout=subprocess.PIPE)
```

Comment:
The URL should not be hard-coded, please make an additional CLI.

Reply:
@brenzi  should the url be interchangeable? and entered per cli?

Reply:
I agree that the url should not be hardcoded. I'd rather have a cli argument for the url and another one for the api key. if both not present, use localhost, if api key not present, assume it isn'nt needed

Reply:
but, actually, the url and  api key should also be searched for in env variables, so you don't need to specify it explicitly each time. (override with cli should be possible though

Reply:
But what would I do if I want to to run the script against a local node? Then I'd need to overwrite the env var to be empty?

I don't have much practice with env vars, but I think I'd prefer having a CLI interface unless you say env vars is much better.

Reply:
we could do cli only, but this clutters the console with redundant info in manual use.
It is quite common to use env for such things. if you want to force localhost you can always override env with cli
My preference in this case is env+cli

Reply:
@brenzi so if I understand you correctly, I need you to add the following env variable:
IPFS_API_ADD: "https://ipfs.infura.io:5001/api/v0/add"